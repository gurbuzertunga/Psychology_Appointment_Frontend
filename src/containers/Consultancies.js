import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import ConsultancyList from '../components/consultancyList';
import AppointmentForm from '../components/appointmentForm';

const Consultancies = props => {
  const { authToken } = props;
  if (!authToken) {
    return <Redirect to="/" />;
  }
  const handleClick = data => console.log(data); // eslint-disable-line
  return (
    <div className="flex lg:flex-row flex-col justify-evenly mt-24">
      <div className="flex flex-col items-center">
        <h1 className="text-center my-4 text-white lg:text-2xl sm:text-4xl text-2xl font-extrabold">Services</h1>
        <ConsultancyList handleClick={handleClick} />
      </div>
      <AppointmentForm />
    </div>
  );
};

Consultancies.propTypes = {
  authToken: PropTypes.string,
};
Consultancies.defaultProps = {
  authToken: '',
};
const mapStateToProps = state => ({
  authToken: state.authenticationReducer.authToken,
});

export default connect(mapStateToProps)(Consultancies);
