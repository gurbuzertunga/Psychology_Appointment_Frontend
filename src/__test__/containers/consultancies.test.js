import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Consultancies from '../../containers/Consultancies';

configure({ adapter: new Adapter() });
const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <Consultancies />
  </Provider>,
);

describe('Consultancies Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });
  test('renders the Consultancies List Component', () => {
    const page = component.find('.consultanciesList');
    expect(page).toMatchSnapshot();
  });
});
