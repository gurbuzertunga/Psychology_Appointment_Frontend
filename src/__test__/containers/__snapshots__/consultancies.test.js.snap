// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consultancies Component renders the Consultancies List Component 1`] = `ShallowWrapper {}`;
